pytmx/pytmx.py

```diff
@@ -395,14 +395,36 @@ def parse_xml(self, node):
 
         # ***         do not change this load order!         *** #
         # ***    gid mapping errors will occur if changed    *** #
-        for subnode in node.findall('layer'):
+        def find_all_visible_nodes(node, node_type):
+            """ Recursively find all visible nodes of the specified type
+
+            Recurses though the ElementTree of a TMX file to find visible nodes
+            by node type.  The layers can be organized in a tree structure of
+            group layers.  This method recurses through any groups to find all
+            such layers in the TMX file.
+
+            :param node: ElementTree xml node
+            :param node_type: Tag name of elements to find
+            :return: Python list of ElementTree xml nodes
+            """
+            subnodes = []
+            for subnode in node:
+                if 'visible' in subnode.attrib and not convert_to_bool(subnode.attrib['visible']):
+                    continue
+                if subnode.tag == node_type:
+                    subnodes.append(subnode)
+                elif subnode.tag == 'group':
+                    subnodes += find_all_visible_nodes(subnode, node_type)
+            return subnodes
+
+        for subnode in find_all_visible_nodes(node, 'layer'):
             self.add_layer(TiledTileLayer(self, subnode))
```

Comment:
This only adds visible layers while previously non-visible layers were also added.  If there is utility in also adding the non-visible layers, it has been lost.  It could be restored with additional changes.

Reply:
I don't think we should be ignoring invisible layers.  For instance, a map may have a shape layer for boundaries that might be invisible on tiled, but we still want it loaded to get the shapes.